google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -195,42 +194,35 @@ def writes_for_set doc_path, data, merge: nil
                 end
                 allow_empty = false
               end
-              return writes_for_set_merge doc_path, data, field_paths, allow_empty
+              return write_for_set_merge doc_path, data, field_paths, allow_empty
             end
 
-            writes = []
-
             data, delete_paths = remove_field_value_from data, :delete
             if delete_paths.any?
               raise ArgumentError, "DELETE not allowed on set"
             end
 
             data, field_paths_and_values = remove_field_value_from data
 
-            writes << Google::Cloud::Firestore::V1::Write.new(
-              update: Google::Cloud::Firestore::V1::Document.new(
-                name: doc_path,
-                fields: hash_to_fields(data))
+            doc = Google::Cloud::Firestore::V1::Document.new(
+              name:   doc_path,
+              fields: hash_to_fields(data)
+            )
+            Google::Cloud::Firestore::V1::Write.new(
+              update:            doc,
+              update_transforms: field_transforms(field_paths_and_values)
             )
```

Comment:
@schmidt-sebastian Please review lines 207-214 to confirm that the `Write` looks correct for a set operation.

Reply:
LGTM